test/integration/action_execution.go

```diff
@@ -639,33 +678,88 @@ type mockProgressTrack struct{}
 func (p *mockProgressTrack) UpdateProgress(actionState proto.ActionResponseState, description string, progress int32) {
 }
 
-func newResizeActionExecutionDTO(actionType proto.ActionItemDTO_ActionType, targetSE, newHostSE, currentSE *proto.EntityDTO) *proto.ActionExecutionDTO {
-	ai := &proto.ActionItemDTO{}
-	ai.TargetSE = targetSE
-	ai.NewSE = newHostSE
-	ai.ActionType = &actionType
+func newResizeActionExecutionDTO(actionType proto.ActionItemDTO_ActionType, targetSE *proto.EntityDTO, changeType int) *proto.ActionExecutionDTO {
+	dto := &proto.ActionExecutionDTO{}
 
-	commType := proto.CommodityDTO_VCPU
-	cap := float64(cpuIncrement)
-	ai.CurrentComm = &proto.CommodityDTO{
-		CommodityType: &commType,
-		Capacity:      &cap,
-	}
+	switch changeType {
+	case REQUEST_SINGLE_CONTAINER:
+		currentSE := newContainerEntity("test-cont")
+		// Build action item on Cpu
```

Comment:
You could refactor the code to build an action item.  There are four almost identical code blocks.

Reply:
Add a new function `newActionItemDTO` to build the action item,  please have a look at my second commit